### PR TITLE
Make it easier to run mwmbl locally

### LINKED ISSUE #48: Unable to run mwmbl locally / index.tinysearch corrupt?
Hi everyone,

I'm trying to run a local development env for mwmbl, but I'm experiencing some difficulties.  
Been following the dev guide: https://github.com/mwmbl/mwmbl/wiki/Development-FAQ

## Steps I've done:
1. Cloning the repo.
```bash
git clone git@github.com:raypatterson77/mwmbl.git
```
3. Setting up Python venv
```bash
python -m venv .     
```
5. Activating Python venv:
```bash
 source bin/activate       
```
7. Install dependencies
```bash
 pip install .        
```
9. Downloaded index File (serveral times) and placed it in data folder
10. Trying to run mwmbl: 
```bash
mwmbl-tinysearchengine --config config/tinysearchengine.yaml  
```
## Errors:
Running
```bash
mwmbl-tinysearchengine --config config/tinysearchengine.yaml  
```
Gives the error:
```bash
usage: mwmbl-tinysearchengine [-h] --index INDEX --terms TERMS
mwmbl-tinysearchengine: error: the following arguments are required: --index, --terms
```
NOTE: /config/tinysearchengine.yaml is not in the master branch, but still available in the update-readme-for-new-crawler branch. Even with /config/tinysearchengine.yaml present, 
```bash
mwmbl-tinysearchengine --config config/tinysearchengine.yaml  
``` 
is failing with the error above.

Trying to use --index --terms paramters as following: 
```bash
mwmbl-tinysearchengine --index data/index.tinysearch --terms data/terms.csv  
```
NOTE: the terms CSV file is nowhere present, not sure about what exactly is expected to be in the file, but from the errors I was getting, 
```bash
term,count
```
needs to be present.  Creating the file terms.csv under data/terms.csv with above content  gives me the error:
```bash
Terms [] []
Traceback (most recent call last):
  File "/home/o/Dokumente/code/python/mwmbl-dev/bin/mwmbl-tinysearchengine", line 8, in <module>
    sys.exit(main())
  File "/home/o/Dokumente/code/python/mwmbl-dev/lib/python3.10/site-packages/mwmbl/tinysearchengine/app.py", line 39, in main
    with TinyIndex(item_factory=Document, index_path=args.index) as tiny_index:
  File "/home/o/Dokumente/code/python/mwmbl-dev/lib/python3.10/site-packages/mwmbl/tinysearchengine/indexer.py", line 81, in __init__
    metadata = TinyIndexMetadata.from_bytes(metadata_bytes)
  File "/home/o/Dokumente/code/python/mwmbl-dev/lib/python3.10/site-packages/mwmbl/tinysearchengine/indexer.py", line 51, in from_bytes
    raise ValueError("This doesn't seem to be an index file")
ValueError: This doesn't seem to be an index file
```

There is no sha sum to validate the integrity of the index.tinysearch file, but as said I downloaded it multiple times and I don't think the file gets corrupted during the download process. 

Am I missing something or is there a problem with the resent version of mwmbl? 
And is there anywhere a example of the terms CSV file?
Is config/tinysearchengine.yaml  for some reason no longer in the master branch? If not I can make a pull requests to add it back from the update-readme-for-new-crawler branch

### Docker
Trying to set up the dev env with docker, ends in similar problems. Unluckily I haven't it documented well, but for building the image config/tinysearchengine.yaml is necessary, as defined in the Dockerfile.
After building: 
```bash
sudo docker run -p 8080:8080 mwmbl                                                                                                                                                                        
usage: mwmbl-tinysearchengine [-h] --index INDEX --terms TERMS
mwmbl-tinysearchengine: error: the following arguments are required: --index, --terms
``` 

### Env
``` bash
uname -a                                                                                                                                                                                                   
Linux o 5.16.7-arch1-1

python --version                                                                                                                                                                                          
Python 3.10.2

pip --version                                                                                                                                                                                                 
pip 21.2.4

docker version                                                                                                                                                                                                
Client:
 Version:           20.10.12
 API version:       1.41
 Go version:        go1.17.5
 Git commit:        e91ed5707e
 Built:             Mon Dec 13 22:31:40 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true
```

### PR DESCRIPTION
**Pull request by daoudclarke**
Fixes #48

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -41,4 +41,4 @@ VOLUME ["/data"]
 EXPOSE 5000
```
Comment (omasanori) [inline Dockerfile:41]: Please copy `devdata/index-v2.tinysearch` to somewhere in the image (`/data`?) for the moment. In long term, we may revisit https://github.com/mwmbl/mwmbl/issues/27 for a better workflow.

File: Dockerfile [hunk 2]
```diff
@@ -41,4 +41,4 @@ VOLUME ["/data"]
...
 
 # Using the mwmbl-tinysearchengine binary/entrypoint which comes packaged with mwmbl
-CMD ["/venv/bin/mwmbl-tinysearchengine"]
+CMD ["/venv/bin/mwmbl-tinysearchengine", "--num-pages", "10240000", "--background"]
```
Comment (omasanori) [inline Dockerfile:44]: Add `--data /data` here if `index-v2.tinysearch` is placed into there.

File: README.md [hunk 3]
```diff
@@ -120,15 +120,27 @@ Development
 ===========
 
 ### Using Docker
+
+WARNING: this uses production settings and creates a **very** large index (around 40Gb)
```
Comment (omasanori) [inline README.md:124]: It is fine for now, I think.

File: README.md [hunk 4]
```diff
@@ -120,15 +120,27 @@ Development
...
+
 1. Create a new folder called `data` in the root of the repository
-2. Download the [index file](https://storage.googleapis.com/mwmbl/index.tinysearch) and place it the new data folder
-3. Run `$ docker build . -t mwmbl`
-4. Run `$ docker run -p 8080:8080 mwmbl`
+2. Run `$ docker build . -t mwmbl`
+3. Run `$ docker run -p 8080:8080 mwmbl`
```
Comment (omasanori) [inline README.md:128]: Don't we need PostgreSQL on Docker setup, by the way?

**Review by omasanori — CHANGES_REQUESTED** [review]
Thank you so much for your consideration! I tested Docker setup but unfortunately it still fails.

File: Dockerfile (hunk 1 above)
Comment (omasanori) [reply Dockerfile:41]: Adding `ADD devdata/index-v2.tinysearch /app/devdata/` seems fine.

**Comment by omasanori** [comment]
Simply requiring Dokku as described at https://github.com/mwmbl/mwmbl/wiki/Deployment might be easier for now.

**Comment by daoudclarke** [comment]
Merging this now, any further issues can be dealt with in a separate PR.

**Pull request merged**